x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/history/ILMHistoryStore.java

```diff
@@ -150,7 +150,7 @@ public void putAsync(ILMHistoryItem item) {
             // TODO: remove the threadpool wrapping when the .add call is non-blocking
             // (it can currently execute the bulk request occasionally)
             // see: https://github.com/elastic/elasticsearch/issues/50440
-            threadPool.executor(ThreadPool.Names.GENERIC).execute(() -> {
+            threadPool.executor(ThreadPool.Names.MANAGEMENT).execute(() -> {
```

Comment:
I thought about this some more and looked at some recent SDHs. We already have a bit of an issue with slow `MANAGEMENT` threads from stats.
I think, even though this is a theoretically more correct. fix as we discussed, this has quite some potential to end weird (thinking about submitting 10k of these tasks in a hot loop and 5 of them blocking the management pool forever if the data node is slow for some reason).
-> I'd say, lets just stick with generic here and assume that bulk size limit increase will be all that's needed.
Maybe we can go for a configurable value here via an undocumented system property just so we have an out n case this causes trouble or is not enough in some corner case?

Reply:
2574688 and aa5be68